omnigibson/robots/r1.py

```diff
@@ -0,0 +1,519 @@
+        We need to cache and restore the base joints to the world.
+        """
+        base_joint_positions = self.get_joint_positions()[self.base_idx]
+        super().reset()
+        self.set_joint_positions(base_joint_positions, indices=self.base_idx)
+
+    def _post_load(self):
+        super()._post_load()
+        # The eef gripper links should be visual-only. They only contain a "ghost" box volume for detecting objects
+        # inside the gripper, in order to activate attachments (AG for Cloths).
+        for arm in self.arm_names:
+            self.eef_links[arm].visual_only = True
+            self.eef_links[arm].visible = False
+
+        self._world_base_fixed_joint_prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(
+            f"{self.prim_path}/rootJoint"
+        )
+        position, orientation = self.get_position_orientation()
+        # Set the world-to-base fixed joint to be at the robot's current pose
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localRot0").Set(
+            lazy.pxr.Gf.Quatf(*orientation[[3, 0, 1, 2]].tolist())
+        )
+
+    # Name of the actual root link that we are interested in. Note that this is different from self.root_link_name,
+    # which is "base_footprint_x", corresponding to the first of the 6 1DoF joints to control the base.
+    @property
+    def base_footprint_link_name(self):
+        return "base_link"
+
+    def _postprocess_control(self, control, control_type):
+        # Run super method first
+        u_vec, u_type_vec = super()._postprocess_control(control=control, control_type=control_type)
+
+        # Change the control from base_footprint_link ("base_footprint") frame to root_link ("base_footprint_x") frame
+        base_orn = self.base_footprint_link.get_orientation()
+        root_link_orn = self.root_link.get_orientation()
+
+        cur_orn = T.mat2quat(T.quat2mat(root_link_orn).T @ T.quat2mat(base_orn))
+
+        # Rotate the linear and angular velocity to the desired frame
+        lin_vel_global, _ = T.pose_transform(th.zeros(3), cur_orn, u_vec[self.base_idx[:3]], th.tensor([0, 0, 0, 1]))
+        ang_vel_global, _ = T.pose_transform(th.zeros(3), cur_orn, u_vec[self.base_idx[3:]], th.tensor([0, 0, 0, 1]))
+
+        u_vec[self.base_control_idx] = th.tensor([lin_vel_global[0], lin_vel_global[1], ang_vel_global[2]])
+        return u_vec, u_type_vec
+
+    def _get_proprioception_dict(self):
+        dic = super()._get_proprioception_dict()
+
+        # Add trunk info
+        joint_positions = ControllableObjectViewAPI.get_joint_positions(self.articulation_root_path)
+        joint_velocities = ControllableObjectViewAPI.get_joint_velocities(self.articulation_root_path)
+        dic["trunk_qpos"] = joint_positions[self.trunk_control_idx]
+        dic["trunk_qvel"] = joint_velocities[self.trunk_control_idx]
```

Comment:
Should we also make a `ArticulatedTrunkRobot`? `_get_proprioception_dict`, `default_proprio_obs`, as well as the logic related to `rigid_trunk` should be moved there.

@cremebrule Also, the way we handle `if self.rigid_trunk` is different for Tiago and Fetch? Can we unify? I felt like the Fetch's way might be the right way.